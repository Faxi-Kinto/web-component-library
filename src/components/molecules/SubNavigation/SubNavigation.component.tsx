import React from 'react';
import * as Styled from './SubNavigation.styles';
import { NavLink } from './';

/**
 * @name SubNavigation
 *
 * Straightforward subNavigation
 *
 * @returns {JSX}
 */
export type SubNavigationProps = {
  links: NavLink[];
  secondary?: boolean;
  vertical?: boolean;
  colorType?: 'light' | 'dark';
};

const SubNavigation: React.FC<SubNavigationProps> = (
  props: SubNavigationProps
): JSX.Element => {
  const { links, secondary, vertical, colorType } = props;
  return (
    <Styled.Nav
      className={`nav${secondary ? ' nav--secondary' : ''}${
        vertical ? ' nav--vertical' : ''
      }${colorType ? ` nav--${colorType}` : ''}`}
    >
      {links.map((link, index) => (
        <a key={index} href={link.route}>
          <span>{link.name}</span>
          {index !== links.length - 1 ? (
            <span className="nav__delimiter">|</span>
          ) : null}
        </a>
      ))}
    </Styled.Nav>
  );
};

export default SubNavigation;
