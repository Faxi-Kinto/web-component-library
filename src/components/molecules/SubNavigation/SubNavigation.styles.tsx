import styled, { css } from 'styled-components';
import { theme } from 'config';
import { pxToRem } from 'styles/fonts';
import { phablet } from 'styles/breakpoints';
import { flexColumn } from 'styles/flex';
import { marginChildren } from 'styles/spacings';
import { listView } from 'styles/containers';

export const Nav = styled.nav`
  ${phablet(css`
    ${flexColumn('center', 'flex-start')};
    ${marginChildren(`0 0 ${pxToRem('24px')} 0`)};
  `)};
  > a {
    color: ${theme.pallet.LIGHT_GRAY};
    font-weight: 500;
    text-decoration: none;

    span {
      &:first-child {
        padding-right: 6px;
      }
    }
    &:not(:last-child) {
      margin-right: 6px;
    }
  }

  &.nav {
    &--light {
      > a {
        color: ${theme.pallet.WHITE};
      }
    }

    &--dark {
      > a {
        color: ${theme.pallet.GRAY_BLUE};
      }
    }

    &--vertical {
      ${listView(pxToRem('15px'), 'center', 'center')};
      text-decoration: underline;
      font-size: ${pxToRem('18px')};

      > a {
        font-weight: inherit;
      }

      .nav__delimiter {
        display: none;
      }
    }

    &--secondary {
      a {
        font-weight: normal;

        span {
          font-size: ${pxToRem('14px')};
          color: ${theme.pallet.LIGHT_GRAY};
        }
      }
    }
  }

  .nav__delimiter {
    ${phablet(css`
      display: none;
    `)};
  }
`;
