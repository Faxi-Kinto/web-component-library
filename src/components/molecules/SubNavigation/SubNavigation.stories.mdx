import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import { theme } from '../../../config';

import SubNavigation from './';

<Meta title="Core|Molecules/SubNavigation" />

# SubNavigation Component

It contains links, each follows you to the desired route.

<!-- TODO ADD STORYBOOK ACTIONS INSTEAD OF ROUTING HERE -->

## Implementation

#### Default

<Preview style={{background: theme.pallet.LIGHT_BLUE}}>
  <Story name="default"
   parameters={{backgrounds: [{ name: 'defaultThemeBackground', value: theme.pallet.LIGHT_BLUE, default: true }]}}
  >
    <SubNavigation
      links={[
        { name: 'About', route: '' },
        { name: 'Pricing', route: '' },
        { name: 'FAQs', route: '' },
        { name: 'News', route: '' },
        { name: 'Contact us', route: '' },
      ]}
    />
  </Story>
</Preview>

#### Dark

<Preview style={{background: theme.pallet.LIGHT_BLUE}}>
  <Story name="secondary"
   parameters={{backgrounds: [{ name: 'defaultThemeBackground', value: theme.pallet.LIGHT_BLUE, default: true }]}}
  >
    <SubNavigation
      secondary
      links={[
        { name: 'Privacy Policy', route: '' },
        { name: 'Terms & Conditions', route: '' },
        { name: 'Cookie Policy', route: '' },
      ]}
    />
  </Story>
</Preview>

## Props

<Props of={SubNavigation} />
