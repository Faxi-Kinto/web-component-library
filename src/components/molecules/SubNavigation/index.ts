import SubNavigation from './SubNavigation.component';

export type NavLink = {
  name: string;
  route: string;
};

export default SubNavigation;
