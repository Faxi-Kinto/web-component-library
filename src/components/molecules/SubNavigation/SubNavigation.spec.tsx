import React from 'react';
import renderer from 'react-test-renderer';
import SubNavigation from './SubNavigation.component';
import { NavLink } from './';

test('SubNavigation renders correctly', () => {
  const data = [
    { name: 'Test 1', route: '' },
    { name: 'Test 2', route: '' },
    { name: 'Test 3', route: '' },
    { name: 'Test 4', route: '' },
  ] as NavLink[];

  const component = renderer.create(<SubNavigation links={data} />);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
