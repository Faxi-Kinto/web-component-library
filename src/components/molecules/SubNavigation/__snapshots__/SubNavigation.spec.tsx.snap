// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubNavigation renders correctly 1`] = `
.c0 > a {
  color: #cdd5d8;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 > a span:first-child {
  padding-right: 6px;
}

.c0 > a:not(:last-child) {
  margin-right: 6px;
}

.c0.nav--light > a {
  color: #FFF;
}

.c0.nav--dark > a {
  color: #4E606A;
}

.c0.nav--vertical {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-size: 1.125rem;
}

.c0.nav--vertical > :first-child {
  max-width: 100%;
  width: 100%;
}

.c0.nav--vertical > * {
  margin: 0 0 0.9375rem 0;
}

.c0.nav--vertical > :last-child {
  margin: 0;
}

.c0.nav--vertical > a {
  font-weight: inherit;
}

.c0.nav--vertical .nav__delimiter {
  display: none;
}

.c0.nav--secondary a {
  font-weight: normal;
}

.c0.nav--secondary a span {
  font-size: 0.875rem;
  color: #cdd5d8;
}

@media (max-width:767px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }

  .c0 > * {
    margin: 0 0 1.5rem 0;
  }

  .c0 > :last-child {
    margin: 0;
  }
}

@media (max-width:767px) {
  .c0 .nav__delimiter {
    display: none;
  }
}

<nav
  className="c0 nav"
>
  <a
    href=""
  >
    <span>
      Test 1
    </span>
    <span
      className="nav__delimiter"
    >
      |
    </span>
  </a>
  <a
    href=""
  >
    <span>
      Test 2
    </span>
    <span
      className="nav__delimiter"
    >
      |
    </span>
  </a>
  <a
    href=""
  >
    <span>
      Test 3
    </span>
    <span
      className="nav__delimiter"
    >
      |
    </span>
  </a>
  <a
    href=""
  >
    <span>
      Test 4
    </span>
  </a>
</nav>
`;
